File: pkg/fingerprint/dicts/qc.py
import json

# 读取 json 文件
with open('file.json', 'r') as f:
    data = json.load(f)

# 去重
unique_data = []
for i in range(len(data)):
    unique = True
    for j in range(i+1, len(data)):
        if data[i]['cms'] == data[j]['cms'] and data[i]['keyword'] == data[j]['keyword']:
            unique = False
            break
    if unique:
        unique_data.append(data[i])

# 输出去重后的结果
print(unique_data)

# 将去重后的结果写入新的 json 文件
with open('unique_file.json', 'w') as f:
    json.dump(unique_data, f)

File: pkg/fingerprint/dicts/qc.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
)

type Fingerprint struct {
	Cms      string   `json:"cms"`
	Method   string   `json:"method"`
	Location string   `json:"location"`
	Keyword  []string `json:"keyword"`
}

func main() {
	// 假设这里的 jsonFile 是您要加载的 JSON 文件
	jsonFile := "fingerprint.json"

	// 读取 JSON 文件内容
	data, err := ioutil.ReadFile(jsonFile)
	if err != nil {
		fmt.Println(err)
		return
	}

	// 解析 JSON 内容
	var fp map[string][]Fingerprint
	if err := json.Unmarshal(data, &fp); err != nil {
		fmt.Println(err)
		return
	}

	// 去重相同 cms 和 keyword 的内容
	cmsKeywordMap := make(map[string]map[string]bool)
	result := make([]Fingerprint, 0)
	for _, fpList := range fp["fingerprint"] {
		key := fpList.Cms + "-" + fmt.Sprintf("%v", fpList.Keyword)
		if cmsKeywordMap[fpList.Cms] == nil {
			cmsKeywordMap[fpList.Cms] = make(map[string]bool)
		}
		if !cmsKeywordMap[fpList.Cms][key] {
			cmsKeywordMap[fpList.Cms][key] = true
			result = append(result, fpList)
		}
	}

	// 将结果转换为 JSON 格式并输出
	output, err := json.MarshalIndent(map[string][]Fingerprint{"fingerprint": result}, "", "  ")
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Println(string(output))
	//if err := ioutil.WriteFile("savs.json", output, os.ModePerm); err != nil {
	//	fmt.Println(err)
	//	return
	//}
}
